### PR TITLE
Refactor context and chat_id #363

### PR DESCRIPTION
**Pull request by maeste**

### CODE REVIEW

**Review by fjuma — CHANGES_REQUESTED** [review]

File: tests/wiseagents/agents/test_sequential_coordinator.py [hunk 1]
```diff
@@ -74,12 +74,13 @@ def response_delivered_restart(message: WiseAgentMessage):
...
 
         try:
-            assert "Raleigh" in response
-            assert "North Carolina" in response
-            assert "Atlanta" in response
-            assert "Georgia" in response
+            #assert "Raleigh" in response
+            #assert "North Carolina" in response
+            #assert "Atlanta" in response
+            #assert "Georgia" in response
```
[the remark this answers is not in the document]
Comment (maeste) [reply tests/wiseagents/agents/test_sequential_coordinator.py:80]: Yup...intermittent failure ~50% for different answers from LLM (correct, but rephrased). Maybe we can open an issue for this, or wait the LLM mock

File: tests/wiseagents/test-multiple.yaml [hunk 2]
```diff
@@ -30,8 +30,8 @@ graph_db: !wiseagents.graphdb.Neo4jLangChainWiseAgentGraphDB
   collection_name: test-cli-vector-db
   properties: [ name, type ]
 llm: !wiseagents.llm.OpenaiAPIWiseAgentLLM
-  model_name: Phi-3-mini-4k-instruct-q4.gguf
-  remote_address: http://localhost:8001/v1
+  model_name: "llama3.1"
+  remote_address: "http://localhost:11434/v1"
```
[the remark this answers is not in the document]
Comment (maeste) [reply tests/wiseagents/test-multiple.yaml:34]: Not yet, but they happen with llama3.2 1B. 3.1 8B works

File: tests/wiseagents/test_message.yaml [hunk 3]
```diff
@@ -1,5 +1,4 @@
 !wiseagents.WiseAgentMessage
-_chat_id: '12345'
```
[the remark this answers is not in the document]
Comment (maeste) [reply tests/wiseagents/test_message.yaml:2]: good catch, I'll fix it

File: src/wiseagents/agents/coordinator_wise_agents.py [hunk 4]
```diff
@@ -216,25 +214,27 @@ def handle_request(self, request):
...
-        ctx.set_collaboration_type(chat_id, WiseAgentCollaborationType.PHASED)
-        ctx.set_route_response_to(chat_id, request.sender)
-
+        sub_ctx_name = f'{self.name}.{str(uuid.uuid4())}'
+
+        ctx = WiseAgentRegistry.create_sub_context(request.context_name, sub_ctx_name)
+        ctx.set_collaboration_type(WiseAgentCollaborationType.PHASED)
+        ctx.set_route_response_to(request.sender)
+        logging.debug(f"set_collaboration_type (0) parent context: {WiseAgentRegistry.get_context(request.context_name)} use_redis: {WiseAgentRegistry.get_context(request.context_name)._use_redis}")
+        logging.debug(f"set_collaboration_type (0) Created context: {ctx} use_redis: {ctx._use_redis}, config: {ctx._config}")
```
[the remark this answers is not in the document]
Comment (maeste) [reply src/wiseagents/agents/coordinator_wise_agents.py:223]: Nothing...and I may remove the logging too. They have been used debugging

**Review by fjuma — APPROVED** [review]

**Pull request merged**